x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalLogicalPlanOptimizer.java

```diff
@@ -132,6 +166,77 @@ protected boolean skipExpression(Expression e) {
         }
     }
 
+    /**
+     * The vast majority of aggs ignore null entries - this rule adds a pushable filter for local aggs
+     * to filter this entries out to begin with.
+     * STATS x = min(a), y = sum(b)
```

Comment:
I think this needs a grouping.

Reply:
It doesn't. The filter only gets added if no grouping is present since it can filter them out if all the group values are null.